src/amberscript/parser.cc

```diff
@@ -1307,6 +1330,150 @@ Result Parser::ParsePipelineDepth(Pipeline* pipeline) {
   return ValidateEndOfStatement("DEPTH command");
 }
 
+Result Parser::ParsePipelineStencil(Pipeline* pipeline) {
+  auto token = tokenizer_->NextToken();
+  if (!token->IsIdentifier())
+    return Result("STENCIL missing face");
+
+  bool setFront = false;
+  bool setBack = false;
+
+  if (token->AsString() == "front") {
+    setFront = true;
```

Comment:
Should these be an enum similar to the compare_op and stencil_op?

Reply:
I don't think so, because one or both booleans can be true. With enums we would later need to do this:

```
if (face == kFront || face == kFrontAndBack)
   doFront()
else if (face == kBack || face == kFrontAndBack)
   doBack()
```